enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,560 @@
+This flag is [crucial](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager) for Cloud Controller Manager engagement within the cluster.
+
+Initially, the new "External" platform should be treated similarly to PlatformType "None" by the MCO, and do not set up any cloud-specific flags for the kubelet.
+Then, down the road (during [phase 3](#implementation-phases)), it would be expected for the MCO to use the "External" platform type and its spec as a signal about the underlying platform and cloud controller manager presence and operate accordingly.
+For an explicit signal about the necessity to set `--cloud-provider=external` flag to the kubelet, we will use the `CloudControllerManager` field of the `ExternalPlatformSpec`, which is  
+described in the [API Extensions](#api-extensions) section down below.
+
+#### Cluster Cloud Controller Manager Operator
+
+Responsible for deploying platform-specific Cloud Controller Manager as well as for handling a number of OpenShift's
+specific peculiarities (such as populating proxy settings for CCMs, sync credentials, and so on).
+
+The code from the [library-go](#library-go) using for decision-making about operator engagement.
+In case library-go's [IsCloudProviderExternal](https://github.com/openshift/library-go/blob/e1213f6ec5d10aa4aa8a4cac2780b1dc674c0396/pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go#L154) function
+indicates that cloud provider is external and the operator encounters a platform which it is not aware of it will go into 'Degraded' state.
+
+During the first phases of the "External" platform type enablement, this operator should be just disabled.
+This might be done with changes within the [library-go](#library-go) and further dependency updates or, better,
+by adding a respective check within the operator itself.
+
+In the future, when the delivery mechanism for CCMs will be defined,
+the operator might be engaged for deploying a user-provided cloud controller manager,
+however this is a subject for the upcoming design work.
+
+#### Machine Api Operator
+
+Machine Api Operator is responsible for deploying and maintaining the set of machine-api related controllers, such as
+
+- machineset controller
+- nodelink controller
+- machine health check controller
+- machine controller
+
+From the list above, only the "machine controller" is cloud-provider dependent, however, for now, Machine Api Operator
+won't deploy anything if it encounters "None" or an unrecognized platform type. 
+
+In the future, "External" platform type, in conjunction with an enabled capability,
+would serve as a signal for Machine Api Operator to deploy only provider-agnostic
+controllers, which would leave room for the user to supplement only the machine controller and not to reverse engineer and
+replicate everything that MAO does.
```

Comment:
Would this be specific to `External`?  It seems like the machine-API operator could deploy the generic controllers on any unrecognized platform type, without waiting for `External` to exist.

Reply:
We can teach machineAPI operator to do so, yes.

But it will be a behaviour change for the `None` platform type, which will affect existing clusters (a few controllers will be added after an upgrade).

Reply:
+1 to what @lobziik said, another aspect of this is that we would like a value to use (eg `External`) so that we can gate any following actions around this new platform type. in this manner it will help us to isolate behavior that should only exist when the cluster is using this type of platform. the hope is that when we see support cases come in, if we see platform "External" and we see that, for example, the machine-api has been enabled, then we know that that user/partner is responsible for the machine controller.

Reply:
One of the goals of this EP is `Introduce a somewhat "neutral" platform type, which would serve as a signal about an underlying generic cloud-provider presence`. Changing MachineAPI operator behaviour for the `None` platform type will introduce even more uncertainty about what is the `None` platform is, IMHO. I would like to avoid this.

Reply:
@lobziik and @elmiko , I like explanation in this thread. Did that text make it into the text of the enhancement somewhere?

Reply:
aside from what @lobziik mentioned in the goals, i'm not sure we talk about the new platform type as a gate for the behavior we want to expose, although it's certainly implied. perhaps we should add a sentence to the motivation section talking about how useful having this value will be to ensuring that we limit the behavior, and support, to only those cases that call out for the "external" platform.